core/trino-spi/src/main/java/io/trino/spi/block/SingleRowBlockWriter.java

```diff
@@ -74,14 +68,22 @@ protected Block getRawFieldBlock(int fieldIndex)
         return fieldBlockBuilders[fieldIndex];
     }
 
+    @Override
+    protected int getRowIndex()
+    {
+        return rowIndex;
+    }
+
     @Override
     public long getSizeInBytes()
     {
         long currentBlockBuilderSize = 0;
+        //  getRegionSizeInBytes returns non-zero result for some fieldBlockBuilder (e.g. for LongArrayBlock) for
```

Comment:
Maybe
```
We need to use subtraction in order to compute size because getRegionSizeInBytes(0, 1) returns non-zero result even if field block builder has no position appended
```

Reply:
ok